tiledb/api/src/array/mod.rs

```diff
@@ -1238,4 +1266,79 @@ pub mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn create_enumeration() -> TileDBResult<()> {
+        let test_uri = tiledb_test_utils::get_uri_generator()
+            .map_err(|e| Error::Other(e.to_string()))?;
+
+        let uri = test_uri
+            .with_path("enumeration_test")
+            .map_err(|e| Error::Other(e.to_string()))?;
+
+        let ctx = Context::new().unwrap();
+        let domain = {
+            let dim = DimensionBuilder::new(
+                &ctx,
+                "dim",
+                Datatype::Int32,
+                ([0, 16], 4),
+            )?
+            .build();
+
+            DomainBuilder::new(&ctx)?.add_dimension(dim)?.build()
+        };
+
+        let enmr = EnumerationBuilder::new(
+            &ctx,
+            "flintstones",
+            Datatype::StringUtf8,
+            "fredwilmageorgebetty".as_bytes(),
```

Comment:
It's certainly not a request for changes in this PR but it feels a little wrong that the default cell val num is single but the other arguments here expect something else.

If offsets is `Some` then it should probably infer CellValNum::Var

Reply:
I'll add a story. It looks like Enumeration predates the CellValNum enum so it'll probably spider in the tests a bit. Also has the feel for another good "first story" for onboarding.

Reply:
https://app.shortcut.com/tiledb-inc/story/54981